src/include/distributed/backend_data.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * backend_data.h
```

Comment:
I like backend_data.h, but I'd consider moving some of the functions and datastructures into transaction_identifier.h or such (`DistributedTransactionId`, `UnSetBackendData`, `GetCurrentDistributedTransctionId`)